src/core/tests/pass_manager.cpp

```diff
@@ -77,6 +79,100 @@ bool validate_list(const std::vector<std::shared_ptr<ov::Node>>& nodes) {
 
 }  // namespace
 
+class TestMatcherPassTrue : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("TestMatcherPassTrue");
+    TestMatcherPassTrue() : MatcherPass() {
+        auto any_input = ov::pass::pattern::any_input();
+        ov::matcher_pass_callback callback = [](ov::pass::pattern::Matcher& m) {
+            return true;
+        };
+
+        auto m = std::make_shared<ov::pass::pattern::Matcher>(any_input, "TestMatcherPassTrue");
+        this->register_matcher(m, callback);
+    }
+};
+
+class TestMatcherPassFalse : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("TestMatcherPassFalse");
+    TestMatcherPassFalse() : MatcherPass() {
+        auto any_input = ov::pass::pattern::any_input();
+        ov::matcher_pass_callback callback = [](ov::pass::pattern::Matcher& m) {
+            return false;
+        };
+
+        auto m = std::make_shared<ov::pass::pattern::Matcher>(any_input, "TestMatcherPassFalse");
+        this->register_matcher(m, callback);
+    }
+};
+
+class TestModelPassTrue : public pass::ModelPass {
+public:
+    OPENVINO_RTTI("TestModelPassTrue");
+
+    bool run_on_model(const std::shared_ptr<ov::Model>& f) override {
+        return true;
+    }
+};
+
+class TestModelPassFalse : public pass::ModelPass {
+public:
+    OPENVINO_RTTI("TestModelPassFalse");
+
+    bool run_on_model(const std::shared_ptr<ov::Model>& f) override {
+        return false;
+    }
+};
+
+class TestValidate : public pass::Validate {
+public:
+    OPENVINO_RTTI("TestValidate");
+
+    bool run_on_model(const std::shared_ptr<ov::Model>& f) override {
+        m_applied = true;
+        return pass::Validate::run_on_model(f);
```

Comment:
```suggestion
        if (!m_applied){
            m_applied = pass::Validate::run_on_model(f);
        }
        return m_applied;
```
? 
to not apply if already done

Reply:
actually there is an agreement  that the transformations should `return true` in case of any changes (reconnections, replacements, adding new ops) in ov::Model inside them. So it means that Validate pass always `returns false` because it only revalidates shapes, but the pass is actually applied, so that's why I'm setting m_applied to `true` here

Reply:
Add comment with this explanation?

Reply:
sure, I will